File: app/src/main/java/com/example/dropex/adpaters/CartGVAdapter.java
package com.example.dropex.adpaters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.dropex.R;
import com.example.dropex.models.CartModel;

import java.util.ArrayList;

public class CartGVAdapter extends ArrayAdapter<CartModel> {

    public CartGVAdapter(@NonNull Context context, ArrayList<CartModel> courseModelArrayList) {
        super(context, 0, courseModelArrayList);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        View listItemView = convertView;
        if (listItemView == null) {
            // Layout Inflater inflates each item to be displayed in GridView.
            listItemView = LayoutInflater.from(getContext()).inflate(R.layout.cart_item, parent, false);
        }

        CartModel cartModel = getItem(position);
        TextView productTVName = listItemView.findViewById(R.id.productNameTV);
        ImageView productImageIV = listItemView.findViewById(R.id.productImageIV);
        listItemView.findViewById(R.id.quantitySetter); // To load quantitySetter
        TextView quantityTV = listItemView.findViewById(R.id.quantityTV);
        TextView sizeTV = listItemView.findViewById(R.id.sizeTV);
        TextView totalPriceTV = listItemView.findViewById(R.id.totalPriceTV);

        String sizeText = "Size: " + cartModel.getSize().toString().toUpperCase();

        productTVName.setText(R.string.placeholder);
        productImageIV.setImageResource(R.drawable.sample);
        quantityTV.setText(String.valueOf(cartModel.getQuantity()));
        sizeTV.setText(sizeText);
        totalPriceTV.setText("$3232");
        return listItemView;
    }
}

File: app/src/main/java/com/example/dropex/adpaters/CartGVAdapter.kt
package com.example.dropex.adpaters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import coil.load
import com.example.dropex.R
import com.example.dropex.database.CartRepository
import com.example.dropex.models.CartModel
import com.example.dropex.models.ProductModel
import java.util.Locale

class CartGVAdapter(context: Context, private var confirmed: Boolean, private var cartModelArrayList: ArrayList<CartModel?>, private var productModelArrayList: ArrayList<ProductModel?>) : ArrayAdapter<CartModel?>(context, 0, cartModelArrayList) {
    private var cartRepository = CartRepository()
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var listItemView = convertView
        if (listItemView == null) {
            listItemView = LayoutInflater.from(context).inflate(R.layout.cart_item, parent, false)
        }

        listItemView!!

        val cartModel = getItem(position)
        val productModel = productModelArrayList[position]
        val productTVName = listItemView.findViewById<TextView>(R.id.productNameTV)
        val productImageIV = listItemView.findViewById<ImageView>(R.id.productImageIV)
        val quantityTV = listItemView.findViewById<TextView>(R.id.quantityTV)
        val sizeTV = listItemView.findViewById<TextView>(R.id.sizeTV)
        val totalPriceTV = listItemView.findViewById<TextView>(R.id.totalPriceTV)
        val closeBtn = listItemView.findViewById<ImageView>(R.id.closeBtnIV)

        cartModel!!
        productModel!!

        if (confirmed) {
            closeBtn.visibility = View.GONE
        } else {
            closeBtn.setOnClickListener {
                // Close button
                cartRepository.removeFromCart(cartModel.productId)
                cartModelArrayList.removeAt(position)
                productModelArrayList.removeAt(position)
                notifyDataSetChanged()
                Toast.makeText(context, "Cart item removed", Toast.LENGTH_SHORT).show()
            }
        }

        val priceText = "$" + cartModel.quantity * productModel.price
        val sizeText = "Size: " + cartModel.size.uppercase(Locale.getDefault())
        val quantityText = "Quantity: " + cartModel.quantity.toString().uppercase(Locale.getDefault())

        productTVName.text = productModel.name
        productImageIV.load(productModel.imgUrl)
        totalPriceTV.text = priceText
        quantityTV.text = quantityText
        sizeTV.text = sizeText
        return listItemView
    }
}